orchestrator/careplancontributor/service.go

```diff
@@ -635,6 +644,29 @@ func (s Service) handleNotification(ctx context.Context, resource any) error {
 			}
 		} else if err != nil {
 			return err
+		} else {
+			//Otherwise, the Task is valid and can be pushed to the frontend
+			data, err := json.Marshal(task)
+			if err != nil {
+				return err
+			}
+
+			// Check if the Task is a subTask
+			var parentTaskReference string
+			if len(task.PartOf) > 0 {
+				for _, reference := range task.PartOf {
+					if reference.Reference != nil && strings.HasPrefix(*reference.Reference, "Task/") {
+						parentTaskReference = *reference.Reference
+						break
+					}
+				}
+			}
+
+			if parentTaskReference != "" {
+				s.sseService.Publish(parentTaskReference, string(data))
+			} else {
+				s.sseService.Publish(fmt.Sprintf("Task/%s", *task.Id), string(data))
+			}
```

Comment:
If I understand correctly, this effectively only publishes parent/main Tasks? And is that right, since `frontend/lib/store/task-progress-store.ts` states that it receives SSE of both parent and subtasks?

Also note that above, the following is executed:

```
fhirBaseURL, _, err := coolfhir.ParseExternalLiteralReference(resourceUrl, *focusReference.Type)
```

The second (currently ignored) return value gives you a relative reference to the resource (e.g. `Task/123`)

Reply:
Yes that's right, only (sub)Tasks are published. Subtasks will be pushed on the topic of the "primary" task that can be subscribed on. So subscribing to subtasks is currently not possible, as the code assumes the frontend subscribes to the main task. The frontend differs on subtasks, either update the existing or add a new one.

I'll use the relative ref from coolfhir instead of the `fmt.Sprintf("Task/%s", *task.Id)`!